src/sentry/integrations/github_enterprise/webhook.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import hashlib
+import hmac
+import logging
+import six
+
+from django.http import HttpResponse
+from django.utils.crypto import constant_time_compare
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from simplejson import JSONDecodeError
+from sentry.models import Integration
+from sentry.utils import json
+from sentry.integrations.github.webhook import InstallationEventWebhook, InstallationRepositoryEventWebhook, PushEventWebhook, PullRequestEventWebhook
+from .repository import GitHubEnterpriseRepositoryProvider
+from .client import GitHubEnterpriseAppsClient
+
+logger = logging.getLogger('sentry.webhooks')
+
+
+def get_installation_metadata(event):
+    try:
+        integration = Integration.objects.get(external_id=event['installation']['id'])
+    except Integration.DoesNotExist:
+        return
+    return integration.metadata['installation']
+
+
+class GitHubEnterpriseInstallationEventWebhook(InstallationEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+
+class GitHubEnterpriseInstallationRepositoryEventWebhook(InstallationRepositoryEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+    # https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent
+    def _handle(self, event, organization, repo):
+        pass
+
+
+class GitHubEnterprisePushEventWebhook(PushEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+    # https://developer.github.com/v3/activity/events/types/#pushevent
+    def is_anonymous_email(self, email):
+        return email[-25:] == '@users.noreply.github.com'
+
+    def get_external_id(self, username):
+        return 'github_enterprise:%s' % username
+
+    def get_client(self, event):
+        metadata = get_installation_metadata(event)
+        if metadata is None:
+            return HttpResponse(status=400)
```

Comment:
you probably either want to return none here or raise an error and then return the HttpResponse from the actual handler. i think the way it is now, this response will just be assigned to the variable here: https://github.com/getsentry/sentry/pull/8781/files#diff-fe274189ff437ed7688ca44af1268a0bR113 and then you'll hit an error at some point when you actually try to use it as the client